rust/arrow/src/ipc/reader.rs

```diff
@@ -563,6 +709,11 @@ pub struct StreamReader<R: Read> {
     ///
     /// This value is set to `true` the first time the reader's `next()` returns `None`.
     finished: bool,
+
+    /// Optional dictionaries for each schema field.
+    ///
+    /// Dictionaries may be appended to in the streaming format.
+    dictionaries_by_field: Vec<Option<ArrayRef>>,
```

Comment:
@andy-thomason would you be able to help with implementing a dictionary reader for the `StreamReader`? This was committed after you had opened this PR, so I rebased it into your branch